src/components/listItem/ListItem.tsx

```diff
@@ -1,18 +1,88 @@
-import "./ListItem.css";
 import { List } from "../../types/List";
+import bin from "../../assets/bin.svg";
+import { useLists } from "../../contexts/ListsContext";
+import { useEffect, useRef, useState } from "react";
+import "./ListItem.css";
 
 interface ListItemInterface {
   list: List;
   isSelected: boolean;
 }
 
 export function ListItem({ list, isSelected }: ListItemInterface) {
+  const [isEditing, setIsEditing] = useState<boolean>(false);
+  const [newListName, setNewListName] = useState<string>(list.name);
+  const inputRef = useRef<HTMLInputElement | null>(null);
+
+  const { deleteList, updateList } = useLists();
+
+  const handleDelete = () => {
+    deleteList(list.id);
+  };
+
+  const handleEdit = () => {
+    setIsEditing(true);
+  };
+
+  const handleSave = () => {
+    if (newListName.length) updateList(list.id, { ...list, name: newListName });
```

Comment:
```suggestion
    if (newListName.length) { updateList(list.id, { ...list, name: newListName }); }
```
Please use brackets for if and else statements unless it is returnin something. Brackets make it more readable.